lapce-data/src/data.rs

```diff
@@ -133,7 +138,67 @@ impl LapceData {
+        let mut files:IndexMap<PathBuf, Option<LineCol>> = IndexMap::new();
+        let pwd = std::env::current_dir().unwrap_or_default();
+
+        paths
+            .iter()
+            .map(Path::new)
+            .filter(|path| !path.exists())
+            .for_each(|path| {
+                let components = dbg!(path.components());
+                if let Some(Component::Normal(argument)) = dbg!(components.last()) {
+                    if let Some(argument) = argument.to_str() {
+                        let args = argument
+                            .rsplitn(3, ':')
+                            .collect::<SmallVec<[&str; 3]>>();
+                        match args.len() {
+                            1 => {
+                                files.insert(pwd.join(args.last().unwrap_or(&"")), None);
+                            },
+                            2 => {
+                                files.insert(
+                                    pwd.join(args.last().unwrap_or(&"")), 
+                                    Some(
+                                        LineCol{
+                                            line: args
+                                                    .first()
+                                                    .unwrap_or(&"1")
+                                                    .parse()
+                                                    .unwrap_or(1),
+                                            column: 1,
+                                        }
+                                    )
+                                );
+                            },
+                            _ => {
+                                files.insert(
+                                    pwd.join(args.last().unwrap_or(&"")), 
+                                    Some( 
+                                        LineCol{ 
+                                            line: args
+                                                .get(1)
+                                                .unwrap_or(&"1")
+                                                .parse()
+                                                .unwrap_or(1), 
+                                            column: args
+                                                .first()
+                                                .unwrap_or(&"1")
+                                                .parse()
+                                                .unwrap_or(1), 
+                                        } 
+                                    ),
+                                );
+                            },
+                        }
+                    } else {
+                        log::warn!("Failed to convert to str.");
+                    }
+                } else {
+                    log::warn!("Failed to obtain last component of path.");
+                }
+            });
```

Comment:
```rust
        let mut dirs = SmallVec::<[PathBuf; 3]>::new();
        let mut files = SmallVec::<[(PathBuf, Option<LineCol>); 3]>::new();
        let pwd = std::env::current_dir().unwrap_or_default();

        paths.iter().for_each(|path| {
            if let Some(path_type) = Self::resolve_path(path, &pwd) {
                match path_type {
                    PathType::File(file, line_col) => files.push((file, line_col)),
                    PathType::Directory(directory) => dirs.push(directory),
                }
            }
        });
```

Reply:
Again:
`paths: Vec<PathBuf>` should be a pure `cli.paths` without any transformations or canonicalization, so go to `launch()` (https://github.com/lapce/lapce/blob/master/lapce-ui/src/app.rs#L139) and do necessary changes.